bbq/vm/vm.go

```diff
@@ -1230,6 +1233,13 @@ func (vm *VM) StackSize() int {
 	return len(vm.stack)
 }
 
+func (vm *VM) Reset() {
+	vm.stack = nil
+	vm.locals = nil
+	vm.callstack = nil
+	vm.ipStack = nil
```

Comment:
Small performance optimization: Maybe we can just reslice instead of assigning `nil`, to re-use the potentially allocated slices and avoid having to re-allocate later?

Maybe we should assert the stack is empty?

Reply:
Good idea. I'll update.

As for checking the stack size: it may not always empty. For example, if the invocation panicked in the middle of the execution, there would be leftovers in the stack.

Reply:
Updated in https://github.com/onflow/cadence/pull/3831/commits/ab34810fe4b04a978be9318e25ad01b50a9ba2ec